### ISSUE TITLE
UpsertDocumentAsync() hangs when using ConnectionPolicy.PreferredLocations

### ISSUE THREAD

**Issue by chriswill** [Azure/azure-documentdb-dotnet#132]
The issue I've encountered is that UpsertDocumentAsync() fails when you use a Multi-Region DocumentDB database account.  I haven't encountered issues with other commands (Create,Replace,Delete) so far, just Upsert.  Here's a simple code example from a console app:

```
static void Main(string[] args)
        {
            Uri accountEndPoint = new Uri(ConfigurationManager.AppSettings["documentDbUrl"]);
            string accountKey = ConfigurationManager.AppSettings["documentDbKey"];

            ConnectionPolicy connectionPolicy = new ConnectionPolicy();
            connectionPolicy.RetryOptions = new RetryOptions
            {
                MaxRetryAttemptsOnThrottledRequests = 20,
                MaxRetryWaitTimeInSeconds = 2
            };

            //Setting read region selection preference 
            connectionPolicy.PreferredLocations.Add("West US");
            connectionPolicy.PreferredLocations.Add("West Europe");

            DocumentClient documentClient = new DocumentClient(
                accountEndPoint,
                accountKey,
                connectionPolicy);

            documentClient.OpenAsync().ConfigureAwait(false);

            string database = "accounts";
            string collection = "collection1";

            Uri documentCollectionUri = UriFactory.CreateDocumentCollectionUri(database, collection);

            DocumentEntity tokenCacheEntity = new DocumentEntity();
            tokenCacheEntity.DocumentType = DocumentType.Account;
            documentClient.UpsertDocumentAsync(documentCollectionUri, tokenCacheEntity).Wait();
        }
```

As written, this hangs on the UpsertDocumentAsync() line.

However, if you comment out the two lines that set the preferred locations, then the code runs successfully to completion.

```
//connectionPolicy.PreferredLocations.Add("West US");
//connectionPolicy.PreferredLocations.Add("West Europe");
```
If this is found to be a valid issue, would love to get some help to work around.  

p.s. DocumentEntity is just a simple POCO test class:

```
    public class DocumentEntity
    {
        [JsonProperty(PropertyName = "id")]
        public string Id { get; set; }

        [JsonProperty(PropertyName = "documentType")]
        public DocumentType DocumentType { get; set; }

        public DocumentEntity()
        {
            Id = Guid.NewGuid().ToString();
        }
    }
```

**Comment by rnagpal** [comment]
@chriswill Thanks for reporting this issue. I'm able to repro this on my end and will let you know once we have pushed an update with the fix.

**Comment by rnagpal** [comment]
@chriswill We have fixed the issue and it will be part of the 1.9.0.0 release of .NET SDK in next few days.

**Comment by chriswill** [comment]
Thanks, that's great news.  I'll definitely upgrade asap.

**Comment by rnagpal** [comment]
@chriswill This has been fixed with 1.9.0 release of .NET SDK
